cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1117,6 +1130,33 @@ protected void jvmPerformanceOptions(List<EnvVar> envVars) {
         return certs;
     }
 
+    /**
+     * Generate the OwnerReference object to link newly created objects to their parent (the custom resource)
+     *
+     * @return
+     */
+    protected OwnerReference createOwnerReference() {
+        return new OwnerReferenceBuilder()
+                .withApiVersion(ownerApiVersion)
+                .withKind(ownerKind)
+                .withName(cluster)
+                .withUid(ownerUid)
+                .withBlockOwnerDeletion(false)
```

Comment:
Why not blocking CR deletion on dependants deletion? I would expect that the CR is deleted only when all dependants are deleted. Then only concern is that if deleting dependants takes more time and the CO reconcile starts, it find the CR and try to create already deleted dependants. is this the reason why you didn't use block or there is any other reasoning ?

Reply:
Actually ... the blocking deletion means that when you do `oc delete kafka my-cluster` this will be hanging _synchronously_ until all owned resources will be deleted. So I think it has bigger probability of what you described happening.

Without it the CR is gone while the sub-components are still being deleted, so this should not really happen. (or at least the probabilities should be smaller)

Reply:
But in general, I'm fine to change that if there is concensus to do it as blocking. It just didn't seemed as needed.

Reply:
Isn't there's a risk either way? With sync delete if CO pod restarts there would be not lock to prevent the new pod seeing the not-yes-deleted CR and recreating resources which k8s had deleted. With async delete there are races if CR is deleted and recreated soon after while dependent resources still exist.

Reply:
So, I spent some time playing with it. From my view, the actual situation when the think doesn't work well is when you create the resource and delete it immediately - the CO will spent 300 second waiting for the Pods to be ready  while they were already wiped out by the garbage collection. The other situation when you delete and create a new one immediately seemed like something where the CO recovers quite easily.

Reply:
That's kinda stupid, because presumably the watch tells us it was deleted, but we basically throw that info away. I suspect it's not _that_ hard to do, we'd just need a `ConcurrentHashMap` which the watch could write to and which we could poll during a reconciliation to check that the resource still exists. That said it doesn't seem justified to me to do that since this looks like a relatively unusual thing to happen in practice and the cluster ends up in the right state eventually.

Reply:
That was my logic as well ... it is all niche issues from which the CO recovers sooner or later. I would give it a time to see if there are any actual issues and act only then.

BTW: The Operator Framework has AFAIR no real deletion. All is done through OwnerReference.